crates/hyperswitch_connectors/src/connectors/novalnet/transformers.rs

```diff
@@ -125,99 +339,712 @@ impl<F, T> TryFrom<ResponseRouterData<F, NovalnetPaymentsResponse, T, PaymentsRe
+pub struct NovalnetResponseTransactionData {
+    pub amount: u32,
+    pub currency: String,
+    pub date: Option<String>,
+    pub order_no: String,
+    pub payment_data: NovalnetResponsePaymentData,
+    pub payment_type: String,
+    pub status: NovalnetTransactionStatus,
+    pub status_code: u16,
+    pub test_mode: u8,
+    pub tid: u64,
+    pub txn_secret: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(untagged)]
+pub enum NovalnetResponsePaymentData {
+    PaymentCard(NovalnetResponseCard),
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct NovalnetResponseCard {
+    pub card_brand: String,
+    pub card_expiry_month: u8,
+    pub card_expiry_year: u16,
+    pub card_holder: String,
+    pub card_number: String,
+    pub cc_3d: u8,
+    pub last_four: String,
+    pub token: Option<String>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct NovalnetPSyncResponse {
+    pub customer: Option<NovalnetResponseCustomer>,
+    pub merchant: Option<NovalnetResponseMerchant>,
+    pub result: ResultData,
+    pub transaction: Option<NovalnetResponseTransactionData>,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Serialize, Default, Deserialize, Clone)]
+pub enum CaptureType {
+    #[default]
+    PARTIAL,
+    FINAL,
+}
+
+#[derive(Default, Debug, Serialize)]
+pub struct Capture {
+    _type: CaptureType,
+    reference: String,
+}
+#[derive(Default, Debug, Serialize)]
+pub struct NovalnetTransaction {
+    tid: String,
+    amount: Option<StringMinorUnit>,
+    capture: Option<Capture>,
+}
+
+#[derive(Default, Debug, Serialize)]
+pub struct NovalnetCaptureRequest {
+    pub transaction: NovalnetTransaction,
+    pub custom: NovalnetCustom,
+}
+
+impl TryFrom<&NovalnetRouterData<&PaymentsCaptureRouterData>> for NovalnetCaptureRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: &NovalnetRouterData<&PaymentsCaptureRouterData>,
+    ) -> Result<Self, Self::Error> {
+        let capture_type = if item.router_data.request.is_multiple_capture() {
+            CaptureType::PARTIAL
+        } else {
+            CaptureType::FINAL
+        };
+
+        let reference = item
+            .router_data
+            .request
+            .multiple_capture_data
+            .as_ref()
+            .map(|multiple_capture_data| multiple_capture_data.capture_reference.clone());
+
+        let capture: Option<Capture> = reference.map(|reference| Capture {
+            _type: capture_type,
+            reference,
+        });
+
+        let transaction = NovalnetTransaction {
+            tid: item.router_data.request.connector_transaction_id.clone(),
+            capture,
+            amount: Some(item.amount.to_owned()),
+        };
+
+        let custom = NovalnetCustom {
+            lang: item
+                .router_data
+                .request
+                .get_browser_info()?
+                .get_language()?,
+        };
```

Comment:
Can we pass default language as English if `browser_info` or `language` is not present?